orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -519,6 +549,31 @@ def test_changes_input_requires_grad_reinitializes_module_gradient_graph_builder
     model(x)
     assert module_gradient_graph_builder != model._module_gradient_graph_builder
 
+@pytest.mark.parametrize("device", ['cuda'])
+def test_input_requires_grad_backward_creates_input_grad_as_required0(device):
+    N, D_in, H, D_out = 32, 784, 500, 10
+    model = NeuralNetMultiplePositionalArgumentsMultipleOutputs0(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x1 = torch.randn(N, D_in, device=device, requires_grad=True)
+    x2 = torch.randn(N, D_in, device=device, requires_grad=False)
+    y1, y2 = model(x1, x2)
+    s = y2.sum()
+    s.backward()
+    assert x1.grad is not None and x2.grad is None
+
+@pytest.mark.parametrize("x1_requires_grad, x2_requires_grad", [(True, True), (True, False), (False, False), (False, True)])
+def test_input_requires_grad_backward_creates_input_grad_as_required1(x1_requires_grad, x2_requires_grad):
+    N, D_in, H, D_out = 32, 784, 500, 10
+    device = 'cuda'
+    model = NeuralNetMultiplePositionalArgumentsMultipleOutputs1(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x1 = torch.randn(N, D_in, device=device, requires_grad=x1_requires_grad)
+    x2 = torch.randn(N, D_in, device=device, requires_grad=x2_requires_grad)
+    y1, y2 = model(x1, x2)
+    s = y2.sum()
+    s.backward()
+    assert (not x1_requires_grad or x1.grad is not None) and (not x2_requires_grad or x2.grad is not None)
```

Comment:
can we add the asserts on the x1.grad and x2.grad correctness? 

see test_multiple_overlapping_forward_backward_calls for example